docs/content/guides/dagster/re-execution.mdx

```diff
@@ -0,0 +1,49 @@
+---
+title: Re-execution | Dagster
+---
+
+# Re-execution in Dagster
+
+This guide is intended to walk through how to re-execute Dagster pipelines and where the subsequent executions are found within Dagit.
+
+## Motivation
+
+If solids fail or upstream data has changed within a pipeline execution, the pipeline may need to be re-run starting from a particular point. We call this process `re-execution`. 
+
+Imagine a machine learning pipeline with three solids, the first being the most time and resource intensive: training the model, testing the model, and building analyses on the results. If the pipeline fails with the solid that is testing the model, after fixing the root cause it would take much more time to create a new run of the pipeline than to start again from the second step.
+
+With Dagster, the re-execution of parts of the pipeline is grouped with the original run to make it easy to trace. The original pipeline execution metadata is not overwritten, making re-execution a non-destructive operation.
+
+## Example Walk-through
+
+Upon navigating to a particular run or launching an execution within Dagit, the re-execution option is found on the top right.
+
+<Image
+alt="Dagit Re-execution"
+src="/images/concepts/dagit/re-execution.png"
+width={3808}
+height={2414}
+/>
+
+### Re-execution Options
+
+There are several options for launching a re-execution. No matter which one is chosen, the re-executed run is linked to the original run, in the above example this is seen under `download_pipeline`.
+
+- **All Steps**: Re-execute the pipeline run from scratch. This option is most relevant upon testing pipeline runs end to end, if there is a desire to associate different runs together.
+- **Selected Step**: Re-execute the selected step(s). This can be done independent of the step status. For different ways to select multiple steps, see below.
+- **From Selected**: Re-execute the pipeline downstream from the selected steps. If a particular step fails, this option is most relevant if there is a desire to run all downstream steps regardless of status.
+- **From Failure**: Re-execute the pipeline run, skipping steps completed successfully. This is only enabled when the pipeline has failed.
```

Comment:
Can we expand this section to explain individual option with more details? These are what we have in the concept page which explain what each button/option is for, while in a guide, we could expand that content and explain further: use examples to showcase how each option benefits in different cases.

Ideally, we could have one example pipeline (what i'm suggesting above) and a user would want different options based on their needs. For example, here are some example workflows in my mind:
- Selected steps: my pipeline is large but I only wanna execute a subset of it, and I know what the exact steps to execute - which can be done by clicking the boxes in gantt chart view.
- From Selected: I'm just developing this single step and want to make sure its downstream works properly too. "So Dagster, please help me figure out the downstream dependencies and run them"
- From Failure: similar to "From Selected", but even more out-of-box, "hey Dagster, re-run the pipeline from failure so I don't need to figure out the dependencies while I think i fix this failed step."

Reply:
I updated the text as you suggested.

Now that I created the example pipeline, I agree it totally makes sense to do a more detailed walk through with updated screenshots.

I see all the screenshots have the header of `demo.elementl.show` - what's the best way for me to create a screenshot of the `unreliable_pipeline` in Dagit that I created and can run  while being consistent with this styling?

Reply:
Great!
As for the screenshots, it doesn't have to be on elementl.show for individual guides. You can screenshot a local dagit. In terms of consistent styling, we usually use Safari to do screenshots because it has a cleaner window style and no ratio spec at the moment.